App.js

```diff
@@ -0,0 +1,171 @@
+        >
+          Learn React
+        </a>
+      </header>
+    </div>*/
+    <div>
+      <PageTitle />
+      <PageTasks /> 
+    </div>
+  );
+}
+
+export default App;
+
+//class page title
+class PageTitle extends React.Component{
+  render(){
+    return (
+      <div className="PageTitle-div">
+        <h2>To Do List</h2>
+        <p>{new Date().getFullYear() + "-" + (new Date().getMonth() + 1 ) + "-" + new Date().getDate()}</p>
+      </div>
+    )
+  }
+}
+
+//add a new task
+class AddNewTask extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      text: ""
+    }
+    this.InputChange = this.InputChange.bind(this);
+    this.SubmitNewTask = this.SubmitNewTask.bind(this);
+  }
+
+  InputChange(event){
+    this.setState({text: event.target.value});
+  }
+
+  SubmitNewTask(event){
+    this.props.GetNewTask(this.state.text);
+    this.setState({text: ""});
+  }
+
+
+  render(){
+    return (
+      <div className="AddNewTask-box">
+        <span className="AddNewTask-items">Add a new Task:</span>
+        <input placeholder="Type something about your new task..." className="AddNewTask-items" onChange={this.InputChange} value={this.state.text}/> 
+        <button className="AddNewTask-items" onClick={this.SubmitNewTask}>Add</button>
+      </div>
+      )
+  }
+}
+
+//task div
+class PageTasks extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      newtasks:[],
+      donetasks:[]
+    }
+
+    this.AddNewTaskItem = this.AddNewTaskItem.bind(this);
+    this.SetTaskAsDone = this.SetTaskAsDone.bind(this);
+  }
+
+  AddNewTaskItem(text){
+    var addToTaskList = true;
+    this.state.newtasks.forEach(task => {
+      if(task.Content == text){
+        addToTaskList =  false;
+        return;
+      }
+    });
+    if(addToTaskList){
+      this.setState({newtasks: this.state.newtasks.concat({Content: text})});
+    }else{
+      alert("This task already exists!");
+    }
+    
+  }
+
+  SetTaskAsDone(text){
+    //update newTask list
+    var updateNewTaskList = [];
+    this.state.newtasks.forEach(task => {
+      if(task.Content != text){
+        updateNewTaskList.push(task);
+      }
+    });
+    this.setState({newtasks: updateNewTaskList});
+    //update doneTask list
+    this.setState({donetasks: this.state.donetasks.concat({Content: text})});
+  }
+
+  render(){
+    const taskFinishRate = Math.round(this.state.donetasks.length/(this.state.newtasks.length + this.state.donetasks.length) * 10000) / 100 + "%";
+    return (
+      <div>
+        <AddNewTask GetNewTask={this.AddNewTaskItem}/>
+        <div>
+    {(this.state.newtasks.length > 0 || this.state.donetasks.length > 0) && <h4 className="DisplayAllTasks-box">All tasks: (Task finish rate: <span>{taskFinishRate}</span>)</h4>}
+          <DisplayAllTasks TaskList={this.state.newtasks} SetTaskDone={this.SetTaskAsDone} TaskState="New"/>
+          <DisplayAllTasks TaskList={this.state.donetasks} TaskState="Done"/>
```

Comment:
[Need discussion]: I understand the `TaskState` distinguish how tasks are displayed so this `DisplayAllTasks` can be reusable. I am not quite a fan of this way of achieving reusability though.

Please look into 
1. https://reactjs.org/docs/composition-vs-inheritance.html
2. https://reactjs.org/docs/higher-order-components.html#gatsby-focus-wrapper

Think about how you can write `DisplayAllTasks` in some other ways to make is reusable.

We will talk about it when we have a chance to talk.